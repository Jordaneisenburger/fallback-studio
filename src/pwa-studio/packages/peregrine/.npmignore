.eslintrc.js
/lib/**/__{docs,helpers,mocks,tests}__/**
/scripts
