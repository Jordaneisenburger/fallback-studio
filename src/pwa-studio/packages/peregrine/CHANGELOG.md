# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.5.0"></a>
# 0.5.0 (2018-08-13)


### Features

* multicasting REST API client ([#164](https://github.com/magento/pwa-studio/issues/164)) ([2852e14](https://github.com/magento/pwa-studio/commit/2852e14)), closes [#140](https://github.com/magento/pwa-studio/issues/140)
* Simulators for async testing ([#24](https://github.com/magento/pwa-studio/issues/24)) ([a380da9](https://github.com/magento/pwa-studio/commit/a380da9))
