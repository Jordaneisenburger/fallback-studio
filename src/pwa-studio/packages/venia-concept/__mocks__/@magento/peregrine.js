import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/lib');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useEventListener = jest.fn(peregrine.useEventListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);
const useWindowSize = jest.fn(peregrine.useWindowSize);
const useToasts = jest.fn(peregrine.useToasts);
// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useEventListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam,
    useWindowSize,

    useToasts
};
