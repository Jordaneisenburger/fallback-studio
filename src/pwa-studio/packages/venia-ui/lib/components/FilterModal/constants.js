export const filterModes = {
    default: 'default',
    swatch: 'swatch'
};

export const filterLayouts = {
    grid: 'grid',
    list: 'list'
};

export const filterRenderOptions = {
    fashion_color: {
        mode: filterModes.swatch,
        options: {
            layout: filterLayouts.grid,
            searchable: true
        }
    },
    default: {
        mode: filterModes.default,
        options: {}
    }
};
