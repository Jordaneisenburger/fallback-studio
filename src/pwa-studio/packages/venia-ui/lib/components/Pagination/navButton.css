.buttonArrow {
    composes: button from './tile.css';
    display: grid;
    padding-bottom: 1px;
}

.buttonInactive {
    composes: buttonArrow;
    color: #999;
}
