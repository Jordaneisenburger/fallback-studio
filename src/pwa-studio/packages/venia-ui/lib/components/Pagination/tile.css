.button {
    grid-row-start: 1;
    outline: none;
    position: relative;
}

.marker {
    position: absolute;
    width: 100%;
    height: 100%;
    border: 1px solid;
    border-radius: 2px;
    bottom: 0;
}
