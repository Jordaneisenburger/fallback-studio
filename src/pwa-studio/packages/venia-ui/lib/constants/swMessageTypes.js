export const HTML_UPDATE_AVAILABLE = 'HTML_UPDATE_AVAILABLE';
export const PREFETCH_IMAGES = 'PREFETCH_IMAGES';
