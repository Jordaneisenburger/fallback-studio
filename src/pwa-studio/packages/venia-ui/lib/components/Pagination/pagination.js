import React, { useMemo } from 'react';
import { func, number, shape, string } from 'prop-types';
import { usePagination } from '@magento/peregrine/lib/talons/Pagination/usePagination';

import { mergeClasses } from '../../classify';
import defaultClasses from './pagination.css';
import Tile from './tile';
import NavButton from './navButton';
import { navButtons } from './constants';

const Pagination = props => {
    const { currentPage, setPage, totalPages } = props.pageControl;

    const talonProps = usePagination({
        currentPage,
        setPage,
        totalPages
    });

    const {
        handleLeftSkip,
        handleRightSkip,
        handleNavBack,
        handleNavForward,
        isActiveLeft,
        isActiveRight,
        tiles
    } = talonProps;

    const navigationTiles = useMemo(
        () =>
            tiles.map(tileNumber => {
                return (
                    <Tile
                        isActive={tileNumber === currentPage}
                        key={tileNumber}
                        number={tileNumber}
                        onClick={setPage}
                    />
                );
            }),
        [currentPage, tiles, setPage]
    );

    if (totalPages === 1) {
        return null;
    }

    const classes = mergeClasses(defaultClasses, props.classes);

    return (
        <div className={classes.root}>
            <NavButton
                name={navButtons.firstPage.name}
                active={isActiveLeft}
                onClick={handleLeftSkip}
                buttonLabel={navButtons.firstPage.buttonLabel}
            />
            <NavButton
                name={navButtons.prevPage.name}
                active={isActiveLeft}
                onClick={handleNavBack}
                buttonLabel={navButtons.prevPage.buttonLabel}
            />
            {navigationTiles}
            <NavButton
                name={navButtons.nextPage.name}
                active={isActiveRight}
                onClick={handleNavForward}
                buttonLabel={navButtons.nextPage.buttonLabel}
            />
            <NavButton
                name={navButtons.lastPage.name}
                active={isActiveRight}
                onClick={handleRightSkip}
                buttonLabel={navButtons.lastPage.buttonLabel}
            />
        </div>
    );
};

Pagination.propTypes = {
    classes: shape({
        root: string
    }),
    pageControl: shape({
        currentPage: number,
        setPage: func,
        totalPages: number
    }).isRequired
};

export default Pagination;
