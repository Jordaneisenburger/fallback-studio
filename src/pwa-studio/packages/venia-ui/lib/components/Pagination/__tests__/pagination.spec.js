import React from 'react';
import { act } from 'react-test-renderer';
import { createTestInstance, usePagination } from '@magento/peregrine';

import { navButtons } from '../constants';
import Pagination from '../pagination';

jest.mock('../../../classify');
jest.mock('../navButton');

const PaginationWrapper = ({ currentPage = 1, totalPages = 3 }) => {
    const [paginationValues, paginationApi] = usePagination({
        initialPage: currentPage,
        initialTotalPages: totalPages
    });

    const pageControl = {
        currentPage: paginationValues.currentPage || currentPage,
        setPage: jest.spyOn(paginationApi, 'setCurrentPage'),
        updateTotalPages: jest.spyOn(paginationApi, 'setTotalPages'),
        totalPages: paginationValues.totalPages || totalPages
    };

    const classes = {
        root: 'root'
    };

    return <Pagination pageControl={pageControl} classes={classes} />;
};

test('renders when there is more than 1 page', () => {
    const instance = createTestInstance(<PaginationWrapper />);

    expect(instance.toJSON()).toMatchSnapshot();
    expect(instance.root.findByProps({ className: 'root' })).toBeTruthy();
});

test('renders nothing when there is only 1 page', () => {
    const instance = createTestInstance(<PaginationWrapper totalPages={1} />);

    expect(instance.toJSON()).toMatchSnapshot();
    expect(instance.root.findAllByProps({ className: 'root' })).toHaveLength(0);
});

test('tiles set the appropriate page number on click', () => {
    const { root } = createTestInstance(<PaginationWrapper />);

    const button = root.findAllByType('button')[2];

    act(() => {
        button.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).toHaveBeenCalledTimes(1);
    expect(setPageSpy).toHaveBeenCalledWith(3);
});

test('prevPage button decrements page by 1', () => {
    const { root } = createTestInstance(<PaginationWrapper currentPage={2} />);

    const prevPage = root.findByProps({ name: navButtons.prevPage.name });
    act(() => {
        prevPage.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).toHaveBeenCalledTimes(1);
    expect(setPageSpy).toHaveBeenLastCalledWith(1);
});

test('prevPage button does nothing if currentPage is 1', () => {
    const { root } = createTestInstance(<PaginationWrapper />);

    const prevPage = root.findByProps({ name: navButtons.prevPage.name });

    act(() => {
        prevPage.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).not.toHaveBeenCalled();
});

test('nextPage button increments page by 1', () => {
    const { root } = createTestInstance(<PaginationWrapper />);

    const nextPage = root.findByProps({ name: navButtons.nextPage.name });

    act(() => {
        nextPage.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).toHaveBeenCalledTimes(1);
    expect(setPageSpy).toHaveBeenLastCalledWith(2);
});

test('nextPage button does nothing on last page', () => {
    const { root } = createTestInstance(<PaginationWrapper currentPage={3} />);

    const nextPage = root.findByProps({ name: navButtons.nextPage.name });

    act(() => {
        nextPage.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).not.toHaveBeenCalled();
});

test('leftSkip skips toward first page', () => {
    const { root } = createTestInstance(
        <PaginationWrapper currentPage={8} totalPages={10} />
    );

    const firstPage = root.findByProps({ name: navButtons.firstPage.name });

    act(() => {
        firstPage.props.onClick();
    });

    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;
    expect(setPageSpy).toHaveBeenCalledTimes(1);
    expect(setPageSpy).toHaveBeenLastCalledWith(3);
});

test('rightSkip skips toward last page', () => {
    const { root } = createTestInstance(<PaginationWrapper />);

    const lastPage = root.findByProps({ name: navButtons.lastPage.name });
    const setPageSpy = root.findByType(Pagination).props.pageControl.setPage;

    act(() => {
        lastPage.props.onClick();
    });

    expect(setPageSpy).toHaveBeenCalledTimes(1);
    expect(setPageSpy).toHaveBeenLastCalledWith(3);
});
