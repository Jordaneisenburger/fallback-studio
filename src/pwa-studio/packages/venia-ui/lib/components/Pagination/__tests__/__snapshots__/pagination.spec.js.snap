// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing when there is only 1 page 1`] = `null`;

exports[`renders when there is more than 1 page 1`] = `
<div
  className="root"
>
  <button
    className="button"
    onClick={[Function]}
  >
    <div
      className="marker"
    />
    1
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    2
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    3
  </button>
</div>
`;
